packages/components/src/components/tab-nav/tab-nav.tsx

```diff
@@ -171,10 +185,24 @@ export class TabNav {
   }
 
   selectTab(nextTab: HTMLScaleTabHeaderElement) {
+    this.resetTabs();
+    nextTab.selected = true;
+    this.selectPanel(nextTab);
+  }
+
+  selectPanel(nextTab: HTMLScaleTabHeaderElement) {
```

Comment:
Separating the panel selection from the `tabselection` makes the sync harder. I suggest having one single function that selects panel and tab together